controller/src/main/java/org/jboss/as/controller/ModelControllerImpl.java

```diff
@@ -544,7 +544,16 @@ boolean boot(final List<ModelNode> bootList, final OperationMessageHandler handl
                     // stop
                     break;
                 } else {
-                    postExtContext.addBootStep(parsedOp);
+                    if(parsedOp.handler instanceof ParallelBootOperationStepHandler &&
```

Comment:
I'm not sure to understand the condition that triggers this.
This does not apply only to the `!remove` operations but to any operations, right?

Reply:
What I meant is that currently ops will be added to the global ops list. 
With delete I'm removing the add/update ops from the list of ops but I also process child ops? It is those child ops that while being updated are not used by the ParallelBootOperationStepHandler which will only use its initial list